apps/dashboard/app/lib/smart_attributes/attributes/bc_num_nodes.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object with defined options
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::BCNumNodes] the attribute object
+    def self.build_bc_num_nodes(opts = {})
+      Attributes::BcNumNodes.new('bc_num_nodes', opts)
+    end
+  end
+
+  module Attributes
+    class BcNumNodes < Attribute
+      def initialize(id, opts)
+        super(id, opts)
+        @opts = @opts.reverse_merge(min: 1, step: 1)
+      end
+
+      # Value of attribute
+      # @return [String] attribute value
+      def value
+        (opts[:value] || '1').to_s
+      end
+
+      # Type of form widget used for this attribute
+      # @return [String] widget type
+      def widget
+        'number_field'
+      end
+
+      # Form label for this attribute
+      # @param fmt [String, nil] formatting of form label
+      # @return [String] form label
+      def label(fmt: nil)
+        (opts[:label] || 'Number of nodes').to_s
+      end
+
+      # Whether this attribute is required
+      # @return [Boolean] is required
+      def required
+        false
+      end
```

Comment:
Not sure why bc_num_slots is required, but I'd say remove this entirely. Having it hard coded like this means the site can't reconfigure it.

Reply:
okay